net/port_test.go

```diff
@@ -0,0 +1,53 @@
+package net
+
+import (
+	"net"
+	"testing"
+)
+
+func TestFreeTCPPortListener(t *testing.T) {
+	aListener, err := FreeTCPPortListener()
+	if err != nil {
+		t.Fatal(err)
+	}
+	bListener, err := FreeTCPPortListener()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if aListener.Addr().(*net.TCPAddr).Port == bListener.Addr().(*net.TCPAddr).Port {
+		t.Errorf("FreeTCPPortListener() got same port: %v, %v", aListener, bListener)
+	}
+	if aListener.Addr().(*net.TCPAddr).Port == 0 || bListener.Addr().(*net.TCPAddr).Port == 0 {
+		t.Errorf("FreeTCPPortListener() got port 0")
+	}
+}
+
+func TestConcurrentFreeTCPPort(t *testing.T) {
```

Comment:
I don't think in this version this test is really necessary

Reply:
Okay. The reason I had it in there was because some folks thought that this code would race. Although, I guess now that we're not closing the listener inside the function, we're less worried about races. So, removeed.